src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -1245,7 +1246,37 @@ PyMODINIT_FUNC PyInit_lpython_module_)" + fn_name + R"((void) {
         bool is_value_dict = ASR::is_a<ASR::Dict_t>(*m_value_type);
         bool alloc_return_var = false;
         std::string indent(indentation_level*indentation_spaces, ' ');
-        if (ASR::is_a<ASR::Var_t>(*x.m_target)) {
+        if (ASRUtils::is_simd_array(x.m_target)) {
+            this->visit_expr(*x.m_target);
+            target = src;
+            if (ASR::is_a<ASR::Var_t>(*x.m_value) ||
+                    ASR::is_a<ASR::ArraySection_t>(*x.m_value)) {
+                std::string arr_element_type = CUtils::get_c_type_from_ttype_t(
+                    ASRUtils::expr_type(x.m_target));
+                std::string size = std::to_string(ASRUtils::get_fixed_size_of_array(
+                    ASRUtils::expr_type(x.m_target)));
+                std::string value;
+                if (ASR::is_a<ASR::ArraySection_t>(*x.m_value)) {
+                    ASR::ArraySection_t *arr = ASR::down_cast<ASR::ArraySection_t>(x.m_value);
+                    this->visit_expr(*arr->m_v);
+                    value = src;
+                    if(!ASR::is_a<ASR::ArrayBound_t>(*arr->m_args->m_left)) {
+                        this->visit_expr(*arr->m_args->m_left);
+                        int n_dims = ASRUtils::extract_n_dims_from_ttype(arr->m_type) - 1;
+                        value += "->data + (" + src + " - "+ value +"->dims["
+                            + std::to_string(n_dims) +"].lower_bound)";
+                    } else {
+                        value += "->data";
+                    }
+                } else if (ASR::is_a<ASR::Var_t>(*x.m_target)) {
```

Comment:
Should this instead be ?
```suggestion
                } else if (ASR::is_a<ASR::Var_t>(*x.m_value)) {
```

Also, based on the `if`-`else` conditions, I think the above case is unused currently (or is like a dead code)?

Reply:
Good catch. Yup, it should be `x.m_value`.

Reply:
Done.

Reply:
>Done.

Could you specify what part was updated? I can see no change in the above `if` statement. Did you push your changes?

Reply:
Now, it seems updated. Thanks.